gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -0,0 +1,331 @@
+      this.deletions = Lists.newLinkedList(checkNotNull(deletions));
+      return this;
+    }
+
+    /**
+     * Associates a server-assigned id to this {@code ChangeRequest}.
+     */
+    Builder id(String id) {
+      this.id = checkNotNull(id);
+      return this;
+    }
+
+    /**
+     * Sets the time when this {@code ChangeRequest} was started by a server.
+     */
+    Builder startTimeMillis(long startTimeMillis) {
+      this.startTimeMillis = startTimeMillis;
+      return this;
+    }
+
+    /**
+     * Sets the current status of this {@code ChangeRequest}.
+     */
+    Builder status(Status status) {
+      this.status = checkNotNull(status);
+      return this;
+    }
+
+    /**
+     * Creates a {@code ChangeRequest} instance populated by the values associated with this
+     * builder.
+     */
+    public ChangeRequest build() {
+      return new ChangeRequest(this);
+    }
+  }
+
+  private ChangeRequest(Builder builder) {
+    this.additions = builder.additions;
+    this.deletions = builder.deletions;
+    this.id = builder.id;
+    this.startTimeMillis = builder.startTimeMillis;
+    this.status = builder.status;
+  }
+
+  /**
+   * Returns an empty builder for the {@code ChangeRequest} class.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Creates a builder populated with values of this {@code ChangeRequest}.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Returns the list of {@link DnsRecord}s to be added to the zone upon executing this {@code
+   * ChangeRequest}.
+   */
+  public List<DnsRecord> additions() {
+    return additions;
+  }
+
+  /**
+   * Returns the list of {@link DnsRecord}s to be deleted from the zone upon executing this {@code
+   * ChangeRequest}.
+   */
+  public List<DnsRecord> deletions() {
+    return deletions;
+  }
+
+  /**
+   * Returns the id assigned to this {@code ChangeRequest} by the server.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this {@code ChangeRequest} was started by the server.
+   */
+  public Long startTimeMillis() {
+    return startTimeMillis;
+  }
+
+  /**
+   * Returns the status of this {@code ChangeRequest}.
+   */
+  public Status status() {
+    return status;
+  }
+
+  com.google.api.services.dns.model.Change toPb() {
+    com.google.api.services.dns.model.Change pb =
+        new com.google.api.services.dns.model.Change();
+    // set id
+    if (id() != null) {
+      pb.setId(id());
+    }
+    // set timestamp
+    if (startTimeMillis() != null) {
+      pb.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC().print(startTimeMillis()));
+    }
+    // set status
+    if (status() != null) {
+      pb.setStatus(status().status);
+    }
+    // set a list of additions
+    if (additions() != null) {
+      LinkedList<com.google.api.services.dns.model.ResourceRecordSet> additionsPb =
```

Comment:
You can initialize an array list with an expected size, however what we should probably do, is not create any new list
but rather use [`Lists.tranform`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Lists.html#transform(java.util.List,%20com.google.common.base.Function)). see [this](https://github.com/GoogleCloudPlatform/gcloud-java/blob/3f2799def9e05353087d2f5816b93eafef8bce38/gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobStatus.java#L116) as an example.